### PR TITLE
Fixed searchController incorrect error code

### PR DESCRIPTION
**Pull request by arbab-cog**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscanning/service/SearchServiceImpl.java [hunk 1]
```diff
@@ -63,21 +63,24 @@ public SearchResponse retrieveByCCDReference(String ccdReference) {
     @Override
     @Transactional
     public SearchResponse retrieveByDcn(String documentControlNumber) {
-        List<EnvelopeCase> envelopeCases = getEnvelopeCaseByDCN(SearchRequest.searchRequestWith()
-                                                                    .documentControlNumber(
-                                                                        documentControlNumber)
-                                                                    .build());
-        if (envelopeCases == null) {
+        if(isEnvelopePaymentPresent(SearchRequest.searchRequestWith().documentControlNumber(documentControlNumber).build())) {
```
Comment (vigneshpushkaranHMCTS) [inline src/main/java/uk/gov/hmcts/reform/bulkscanning/service/SearchServiceImpl.java:66]: assign to var and pass in the method

File: src/main/java/uk/gov/hmcts/reform/bulkscanning/service/SearchServiceImpl.java [hunk 2]
```diff
@@ -63,21 +63,24 @@ public SearchResponse retrieveByCCDReference(String ccdReference) {
...
+
+            List<EnvelopeCase> envelopeCases = getEnvelopeCaseByDCN(SearchRequest.searchRequestWith()
```
[the remark this answers is not in the document]
Comment (arbab-cog) [reply src/main/java/uk/gov/hmcts/reform/bulkscanning/service/SearchServiceImpl.java]: Changes made

**Review by aravindan89 — APPROVED** [review]

**Review by AnilDhawaleGitHub — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/reform/bulkscanning/service/SearchServiceTest.java [hunk 3]
```diff
@@ -150,4 +152,32 @@ public void testRetrieveByDcn() throws Exception {
         SearchResponse searchResponse = paymentService.retrieveByDcn(TEST_DCN_REFERENCE);
         assertThat(searchResponse.getPayments().get(0).getDcnReference()).isEqualTo(TEST_DCN_REFERENCE);
     }
+
+    @Test
+    @Transactional
```
Comment (YasharRahvar) [inline src/test/java/uk/gov/hmcts/reform/bulkscanning/service/SearchServiceTest.java:157]: why Transactional for unit test?

Comment (arbab-cog) [reply src/test/java/uk/gov/hmcts/reform/bulkscanning/service/SearchServiceTest.java:157]: Removing transactional annotation as these test cases do not interact with the database but use a mocked version of results. There are no changes being made in the database to be reverted back.

**Review by AyanDuttaHMCTS — APPROVED** [review]
Ignoring PMD rules justified as we are currently scanning with Sonar and planning to tie up with Fortify

**Pull request closed without merging**